src/System.Private.CoreLib/shared/System/Exception.cs

```diff
@@ -140,27 +140,21 @@ public virtual void GetObjectData(SerializationInfo info, StreamingContext conte
         }
 
         public override string ToString()
-        {
-            return ToString(true, true);
-        }
-
-        private string ToString(bool needFileLineInfo, bool needMessage)
         {
             string s = GetClassName();
 
-            string? message = (needMessage ? Message : null);
-            if (!string.IsNullOrEmpty(message))
+            if (!string.IsNullOrEmpty(Message))
             {
-                s += ": " + message;
+                s += ": " + Message;
```

Comment:
This could be a breaking change. We used to call `Exception.Message` once but now do it twice. Its a virtual method so could change or throw etc. between calls. I suggest maybe putting `Message` into a local variable?

Reply:
It was in a local. I don't understand what this changed. Seems like the only change needed was to remove the bool check above and just always access Message when initializing message.

Reply:
> This could be a breaking change.

Even if it weren't (and arguably it'd be an acceptable one, as Message should be idempotent), it's more expensive to call it twice: many exception types build up Message via string concatenation and the like.

Reply:
Cached.